library/hmac_drbg.c

```diff
@@ -233,20 +280,15 @@ int mbedtls_hmac_drbg_seed( mbedtls_hmac_drbg_context *ctx,
      *
      * (This also matches the sizes used in the NIST test vectors.)
      */
-    entropy_len = md_size <= 20 ? 16 : /* 160-bits hash -> 128 bits */
-                  md_size <= 28 ? 24 : /* 224-bits hash -> 192 bits */
-                                  32;  /* better (256+) -> 256 bits */
-
-    /*
-     * For initialisation, use more entropy to emulate a nonce
-     * (Again, matches test vectors.)
-     */
-    ctx->entropy_len = entropy_len * 3 / 2;
+    ctx->entropy_len = md_size <= 20 ? 16 : /* 160-bits hash -> 128 bits */
+                       md_size <= 28 ? 24 : /* 224-bits hash -> 192 bits */
+                       32;  /* better (256+) -> 256 bits */
 
-    if( ( ret = mbedtls_hmac_drbg_reseed( ctx, custom, len ) ) != 0 )
+    if( ( ret = hmac_drbg_reseed_core( ctx, custom, len,
+                                       1 /* add nonce */ ) ) != 0 )
+    {
         return( ret );
-
-    ctx->entropy_len = entropy_len;
```

Comment:
Previously, it appears `ctx->entropy_len` was set to a value that included the nonce (32 * 3 / 2 at most). Now, it only includes without nonce (just 32). I looked at `hmac_drbg_reseed_core()` as well to see if it would be setting `ctx->entropy_len`, but it doesn't add the nonce length to `ctx->entropy_len`.

Is it OK that ctx->entropy_len no longer has the nonce length included?

Reply:
@Patater `ctx->entropy_len` was previously set to `3/2 * security` just for the duration of the initial seeding, and was reset to `security` afterwards. Now this single use of factor `3/2` is controlled through the added parameter, while `ctr->entropy_len` is always `security`.

Reply:
OK, I missed that it was reset to `security` length